File: pset1/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;

    // Solicita ao usuário a altura da pirâmide
    do
    {
        printf("Digite a altura da pirâmide (um número inteiro entre 1 e 8): ");
        scanf("%d", &height);

        // Verifica se a altura está dentro do intervalo desejado
        if (height < 1 || height > 8)
        {
            printf("Por favor, insira um número entre 1 e 8.\n");
        }
    }
    while (height < 1 || height > 8);

    // Loop para iterar pelas linhas
    for (int i = 0; i < height; i++)
    {
        // Loop para imprimir espaços em branco antes no meio e depois dos caracteres '#'
        for (int j = 0; j < height - i - 1; j++)
        {
            printf(" ");
        }

        // Loop para imprimir os caracteres duplos '#' na linha atual
        for (int k = 0; k < i + 1; k++)
        {
            printf("#");
        }

        // Adicionar espaços entre as duas pirâmides
        printf("  ");

        // Loop para imprimir os caracteres '#' á direita
        for (int l = 0; l < i + 1; l++)
        {
            printf("#");
        }

        // Quebra de linha após cada linha da pirâmide
        printf("\n");
    }

    return 0;
}

File: pset6/more/mario.py
from cs50 import get_int

# Solicitar altura da meia pirâmide ao usuário
height = get_int("Height: ")

# Verificar se a altura está dentro do intervalo desejado
while height not in range(1, 9):
    print("A altura deve estar entre 1 e 8")
    height = get_int("Height: ")

# Imprimir linhas de hashtags para a meia pirâmide esquerda e meia pirâmide direita
for i in range(height):
    print(" " * (height - i - 1) + "#" * (i + 1) + "  " + "#" * (i + 1))
